R/as_gt.R

```diff
@@ -79,23 +79,37 @@ as_gt <- function(x, ...) {
 #'   summary() %>%
 #'   as_gt()
 as_gt.fixed_design <- function(x, title = NULL, footnote = NULL, ...) {
-  # get the design method
-  design_mtd <- intersect(
-    c("ahr", "fh", "mb", "lf", "rd", "maxcombo", "milestone", "rmst"), class(x)
-  )[1]
+  method <- fixed_method(x)
+  ans <- gt::gt(x) %>%
+    gt::tab_header(title = title %||% fixed_title(method)) %>%
+    gt::tab_footnote(
+      footnote = footnote %||% method_footnote(x, method),
+      locations = gt::cells_title(group = "title")
+    )
+  return(ans)
+}
+
+# get the fixed design method
+fixed_method <- function(x) {
```

Comment:
Is it appropriate to refer to this as the 'fixed_design_method'? From my perspective, the term 'fixed method' implies that the method is fixed, which may have limited clinical significance.

Furthermore, I've observed utility functions for both fixed design (such as `fixed_method`, `fixed_title`, `method_footnote`), as well as group sequential design (such as `get_display_columns `, `footnote_content`, `footnote_non_binding`, `footnote_row`, `gs_design_info`). I am considering whether it would be beneficial to use distinct prefixes for these two sets of utilities to differentiate their purposes.

Finally, I am wondering if we should save the above 2 sets of utilities into a separate R file, and name it as `utility_as_gt`.

Reply:
> I am considering whether it would be beneficial to use distinct prefixes for these two sets of utilities to differentiate their purposes.

I'm using the prefixes `fd_` and `gsd_` now, and am open to other suggestions. I tend not to use the prefix `gs_design_` because I noticed that we have had functions with this prefix but serve a very different purpose. I hope the abbreviations `fd` and `gsd` make sense. They are only for ourselves and users won't use them, so the naming is perhaps not critically important. Anyway, naming is really hard :)

> Finally, I am wondering if we should save the above 2 sets of utilities into a separate R file, and name it as `utility_as_gt`.

I don't have an opinion on that. I can certainly put these functions in `as_utils.R` if you prefer.

Reply:
I like the `fd` and `gsd`, which are sufficiently clear for developers. I am also good for the current folder/file structure. I will get it merged. Thanks for all the work!